b2share/modules/deposit/api.py

```diff
@@ -74,6 +81,9 @@ def build_deposit_schema(self, record):
     @classmethod
     def create(cls, data, id_=None):
         """Create a deposit."""
+        # check that the status field is not set
+        assert 'publication_state' not in data
```

Comment:
Because we have two inputs for the programmatic API.
1/ CLI: It should use the API properly whatever happens.
2/ REST API: it should filter its input using the input loaders. The loaders should be the one raising the error.

Thus if we arrive here with a `publication_state` set there is something wrong going on,
Later we might disable this check as we will have to handle B2Share migration (2.0->2.x).